In [1]:
import pandas as pd
import numpy as np


In [2]:
play = pd.read_csv('googleplaystore.csv')

In [3]:
#Checking data
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
play.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


Total rows 10841

### Quality
* Life Made WI-Fi Touchscreen shifted, data needs to be adjusted

### Tidiness
* Installs has plus symbol that needs to be removed
* Android ver needs to remove and up
* M removed from size
* Reviews, installs, price need to change to floats(int)
* Last updated changed to datetime
* Change Category to title case
* Remove _ from Category


## Incomplete

* Possibly split up Genres for games with multiple genres
* CC Rivals, Type needs to be changed to Free
* Convert size to KB and then float

Current Version Changes
1.1.2 - Learn To Draw Kawaii Characters
4 - Virtual DJ Sound Mixer
1.1.2 - BT Master	
2.17.8 - Dots puzzle

Group Data into categories i.e. low, med, high
Price
Rating

In [5]:
play = play.rename(columns = {'Content Rating': 'Content_Rating', 'Last Updated': 'Last_Updated', 'Current Ver': 'Current_Ver', 'Android Ver': 'Android_Ver'})

In [6]:
play[play['App']== 'Life Made WI-Fi Touchscreen Photo Frame']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content_Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last_Updated Current_Ver Android_Ver  
10472       1.0.19  4.0 and up         NaN

In [7]:
play['Current_Ver'] = play['Current_Ver'].str.replace('4.0 and up', '1.0.19')
play['Last_Updated'] = play['Last_Updated'].str.replace('1.0.19', 'February 11, 2018')
play['Genres'] = play['Genres'].str.replace('February 11, 2018', 'Lifestyle')
play['Content_Rating'] = play['Content_Rating'].fillna('Everyone')
play['Price'] = play['Price'].str.replace('Everyone', 'Free')
play['Type'] = play['Type'].str.replace('0', 'Free')
play['Installs'] = play['Installs'].str.replace('Free', '1000')
play['Size'] = play['Size'].str.replace('1,000+', '3')
play['Size'] = play['Size'].str.replace('+', '')
play['Reviews'] = play['Reviews'].str.replace('3.0M', '19')
play['Rating'] = play['Rating'].replace(19,1.9)
play['Category'] = play['Category'].str.replace('1.9', 'LIFESTYLE')
play[play['App']== 'Life Made WI-Fi Touchscreen Photo Frame']



App   Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame  LIFESTYLE     1.9      19   

      Size Installs  Type Price Content_Rating     Genres       Last_Updated  \
10472    3     1000  Free  Free       Everyone  Lifestyle  February 11, 2018   

      Current_Ver Android_Ver  
10472      1.0.19         NaN

Define:
Several of the columns have values that are not required. The + sign from Installs, the 'and up' from Android Ver, the 'M' from Size. This section makes those changes

In [8]:
#Removing the Plus sign from installs
play['Installs'] = play['Installs'].map(lambda x: x.rstrip('+')).str.replace(',','')

#Removing the 'and up' from Android Version
play['Android_Ver'] = play['Android_Ver'].str.replace('and up','')

play['Size'] = play['Size'].str.replace('M','')

play['Price'] = play['Price'].str.replace('$','')
play['Price'] = play['Price'].str.replace('Free','0')

play.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size  Installs  Type Price  \
10836     4.5      38                  53      5000  Free     0   
10837     5.0       4                 3.6       100  Free     0   
10838     NaN       3                 9.5      1000  Free     0   
10839     4.5     114  Varies with device      1000  Free     0   
10840     4.5  398307                  19  10000000  Free     0   

      Content_Rating             Genres      Last_Updated         Current_Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android_Ver  
10836                4.1   
10837                4.1   
10838                2.2   
10839  Varies with device  
10840  Varies with device

* Reviews, size, installs, price need to change to floats(int)

## Define
Converting datatypes for Reviews, Installs, Price, and Size

## Code

In [9]:
convert_dict = {'Reviews': float, 'Installs': int, 'Price':float}
play = play.astype(convert_dict)
play['Last_Updated'] = play['Last_Updated'].astype('datetime64[ns]')



## Test


In [10]:
play.dtypes

App                       object
Category                  object
Rating                   float64
Reviews                  float64
Size                      object
Installs                   int32
Type                      object
Price                    float64
Content_Rating            object
Genres                    object
Last_Updated      datetime64[ns]
Current_Ver               object
Android_Ver               object
dtype: object

## Define
The titles in categories needs to be changed to title case, the _ also needs to be removed

## CODE

In [11]:
play['Category'] = play['Category'].str.title()
play['Category'] = play['Category'].str.replace('And','and')
play['Category'] = play['Category'].str.replace('_' , ' ')
play['App'] = play['App'].str.title()


In [12]:
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & Scrapbook  Art and Design     4.1   
1                                Coloring Book Moana  Art and Design     3.9   
2  U Launcher Lite – Free Live Cool Themes, Hide ...  Art and Design     4.7   
3                              Sketch - Draw & Paint  Art and Design     4.5   
4              Pixel Draw - Number Art Coloring Book  Art and Design     4.3   

    Reviews Size  Installs  Type  Price Content_Rating  \
0     159.0   19     10000  Free    0.0       Everyone   
1     967.0   14    500000  Free    0.0       Everyone   
2   87510.0  8.7   5000000  Free    0.0       Everyone   
3  215644.0   25  50000000  Free    0.0           Teen   
4     967.0  2.8    100000  Free    0.0       Everyone   

                      Genres Last_Updated         Current_Ver Android_Ver  
0               Art & Design   2018-01-07               1.0.0      4.0.3   
1  Art & Design;Pretend Play   2018-01-15               2.0.0      4.0.3   
2               Art & Design   2018-08-01               1.2.4      4.0.3   
3               Art & Design   2018-06-08  Varies with device        4.2   
4    Art & Design;Creativity   2018-06-20                 1.1        4.4

Univariable Analysis

## Category

In [32]:
play.Category.describe()

count      10841
unique        33
top       Family
freq        1972
Name: Category, dtype: object

In [17]:
play.groupby('Category')['App'].count().sort_values(ascending = False)

Category
Family                 1972
Game                   1144
Tools                   843
Medical                 463
Business                460
Productivity            424
Personalization         392
Communication           387
Sports                  384
Lifestyle               383
Finance                 366
Health and Fitness      341
Photography             335
Social                  295
News and Magazines      283
Shopping                260
Travel and Local        258
Dating                  234
Books and Reference     231
Video Players           175
Education               156
Entertainment           149
Maps and Navigation     137
Food and Drink          127
House and Home           88
Libraries and Demo       85
Auto and Vehicles        85
Weather                  82
Art and Design           65
Events                   64
Comics                   60
Parenting                60
Beauty                   53
Name: App, dtype: int64

## Rating

In [30]:
play.groupby('Rating')['App'].count().sort_values(ascending = False)

Rating
4.4    1109
4.3    1076
4.5    1038
4.2     952
4.6     823
4.1     708
4.0     568
4.7     499
3.9     386
3.8     303
5.0     274
3.7     239
4.8     234
3.6     174
3.5     163
3.4     128
3.3     102
4.9      87
3.0      83
3.1      69
3.2      64
2.9      45
2.8      42
2.7      25
2.6      25
2.5      21
2.3      20
2.4      19
1.0      16
2.2      14
1.9      14
2.0      12
1.8       8
1.7       8
2.1       8
1.6       4
1.5       3
1.4       3
1.2       1
Name: App, dtype: int64

In [31]:
play.Rating.describe()

count    9367.000000
mean        4.191513
std         0.515735
min         1.000000
25%         4.000000
50%         4.300000
75%         4.500000
max         5.000000
Name: Rating, dtype: float64

## Reviews

In [34]:
play.Reviews.describe()

count    1.084100e+04
mean     4.441119e+05
std      2.927629e+06
min      0.000000e+00
25%      3.800000e+01
50%      2.094000e+03
75%      5.476800e+04
max      7.815831e+07
Name: Reviews, dtype: float64

In [35]:
play.Installs.describe()

count    1.084100e+04
mean     1.546291e+07
std      8.502557e+07
min      0.000000e+00
25%      1.000000e+03
50%      1.000000e+05
75%      5.000000e+06
max      1.000000e+09
Name: Installs, dtype: float64

In [51]:
play.sort_values(by='Installs', ascending = False)

App             Category  \
3896                                 Subway Surfers                 Game   
3943                                       Facebook               Social   
335        Messenger – Text And Video Chat For Free        Communication   
3523                                   Google Drive         Productivity   
3565                                   Google Drive         Productivity   
2884                                  Google Photos          Photography   
3665                                        Youtube        Video Players   
2853                                  Google Photos          Photography   
4234                  Skype - Free Im & Video Calls        Communication   
3687                        Google Play Movies & Tv        Video Players   
2808                                  Google Photos          Photography   
4170                                   Google Drive         Productivity   
4153                                       Hangouts        Communication   
4150                                         Google                Tools   
152                               Google Play Books  Books and Reference   
4144                                        Google+               Social   
4104       Messenger – Text And Video Chat For Free        Communication   
4098                      Maps - Navigate & Explore     Travel and Local   
4096                                          Gmail        Communication   
3736                                    Google News   News and Magazines   
3765                                    Google News   News and Magazines   
9844                                    Google News   News and Magazines   
3816                                    Google News   News and Magazines   
336                              Whatsapp Messenger        Communication   
340                                           Gmail        Communication   
341                                        Hangouts        Communication   
464                                        Hangouts        Communication   
5856                              Google Play Games               Family   
1872                                 Subway Surfers                 Game   
5395                                  Google Photos          Photography   
...                                             ...                  ...   
10530                                     Fk Sileks               Sports   
9532              Ek-Yatri: Travel Where You Belong     Travel and Local   
9139                                    Dz Register         Productivity   
9262                                 Ec-Contractors       House and Home   
5812                        Ax Watch For Watchmaker      Personalization   
5921                                       Ba Sales        Communication   
6962                                    Kick Axe Bx   Health and Fitness   
10775                            Fq - Football Quiz               Sports   
10101                                     Amleen Ey               Social   
5917                                       Ra Ga Ba                 Game   
10764                             Fp Transportation    Auto and Vehicles   
10654                   Wallpapers Fn Herstal Fnp 9      Personalization   
4459           Pistachio Launcher For Android P 9.0      Personalization   
6239                       Bh - Fitness & Nutrition   Health and Fitness   
10541                            Fk Bregalnica Štip               Sports   
9934              I'M Rich/Eu Sou Rico/أنا غني/我很有錢            Lifestyle   
4465   Popsicle Launcher For Android P 9.0 Launcher      Personalization   
9719                                   Ep Cook Book              Medical   
9337                       Eg | Explore Folegandros     Travel and Local   
6692                                  Cronometra-Br         Productivity   
8614                              Sweden Newspapers   News and Magazines   
9917                

In [37]:
play.groupby('Type')['App'].count().sort_values(ascending = False)

Type
Free    10040
Paid      800
Name: App, dtype: int64

In [36]:
play.Type.describe()

count     10840
unique        2
top        Free
freq      10040
Name: Type, dtype: object

In [38]:
play.groupby('Price')['App'].count().sort_values(ascending = False)

Price
0.00      10041
0.99        148
2.99        129
1.99         73
4.99         72
3.99         63
1.49         46
5.99         30
2.49         26
9.99         21
6.99         13
399.99       12
14.99        11
4.49          9
24.99         7
3.49          7
7.99          7
29.99         7
5.49          6
19.99         6
11.99         5
12.99         5
8.99          5
6.49          5
10.00         3
2.00          3
16.99         3
1.00          3
10.99         2
8.49          2
          ...  
25.99         1
394.99        1
389.99        1
379.99        1
299.99        1
200.00        1
154.99        1
109.99        1
89.99         1
74.99         1
46.99         1
37.99         1
30.99         1
28.99         1
19.90         1
3.88          1
19.40         1
18.99         1
15.99         1
15.46         1
14.00         1
4.85          1
4.84          1
4.80          1
4.77          1
4.60          1
4.59          1
4.29          1
3.90          1
5.00          1
Name: App, Length:

In [39]:
play.Price.describe()

count    10841.000000
mean         1.027273
std         15.948971
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max        400.000000
Name: Price, dtype: float64